### ISSUE TITLE
Does YOLOv8 classification support FP16 and INT8?

### ISSUE THREAD

**Issue by pornpra** [ultralytics/ultralytics#13373]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have already trained a YOLOv8 classification model (FP32) using Ultralytics version 8.2.18. Does YOLOv8 classification support FP16 and INT8 conversion?

**Comment by glenn-jocher** [comment]
Yes, YOLOv8 classification models do support FP16 and INT8 quantization. You can convert your trained FP32 model to these formats using the `export` mode with the appropriate arguments (`half=True` for FP16 and `int8=True` for INT8). This conversion can help optimize your model for faster inference speeds and reduced model size, suitable for deployment on platforms with limited computational resources.

If you need further guidance on how to perform these conversions, please refer to the export section of our documentation.

**Comment by pornpra** [comment]
@glenn-jocher  Thanks for your answers :)

To export a trained custom YOLOv8 classification model from PyTorch format (FP32) to TensorRT or engine format (INT8), how many steps are involved in the export process, and is it necessary to prepare a calibration dataset for use during the calibration process with TensorRT?

**Comment by glenn-jocher** [comment]
Hello! 😊

Exporting your custom YOLOv8 classification model to TensorRT (INT8) involves a few key steps:

1. **Prepare Calibration Dataset**: Yes, you'll need a representative calibration dataset. This dataset helps TensorRT determine optimal scaling factors for INT8 quantization to maintain accuracy.

2. **Export the Model**: Use the `export` function with the `format='engine'`, `int8=True`, and specify your calibration dataset using the `data` argument.

3. **Calibration Process**: During export, TensorRT will perform the calibration using the provided dataset to optimize the model for INT8 precision.

That's it! These steps will convert your model to the TensorRT format optimized for INT8 operations. If you have any more questions or need further assistance, feel free to ask. Happy modeling! 🚀

**Comment by yunlongwangsd** [comment]
Hello, I've tried to export an `INT8 TensorRT` model using the above method, and the evaluation result is different from the `TensorRT` model without adding `int8=True`. However, when using `yolo val detect` function, there is no difference whether adding `int8=True` or not, but the result is different when adding `half=True`. I am wondering if YOLOv8 classification supports any quantization for default PyTorch model (`yolov8n.pt`) when using `int8` and `half` arguments. Any help would be appreciated!

**Comment by glenn-jocher** [comment]
Hello! 😊

Thank you for your question and for providing detailed information about your experience with exporting the INT8 TensorRT model.

To address your concerns:

1. **INT8 Quantization**: When exporting a model with `int8=True`, TensorRT performs post-training quantization, which can lead to differences in evaluation results compared to FP32 models. This is expected due to the reduced precision. Ensure you have a representative calibration dataset specified using the `data` argument during export to achieve optimal performance.

2. **FP16 Quantization**: Using `half=True` converts the model to FP16 precision, which can also result in different evaluation metrics compared to FP32. This is because FP16 has lower precision than FP32 but offers faster inference speeds.

3. **YOLOv8 Classification Support**: Yes, YOLOv8 classification models do support both FP16 and INT8 quantization. You can export your PyTorch model (`yolov8n.pt`) to TensorRT with these quantizations using the following commands:
    ```python
    from ultralytics import YOLO

    model = YOLO("yolov8n.pt")
    model.export(format="engine", int8=True, data="path/to/calibration_dataset.yaml")
    model.export(format="engine", half=True)
    ```

4. **Validation Differences**: The `yolo val detect` function should reflect the quantization differences. If you observe no difference, ensure you are using the latest version of the Ultralytics package and that the calibration dataset is correctly specified for INT8 quantization.

If the issue persists, please provide a minimum reproducible example to help us diagnose the problem more effectively. You can find guidance on creating a reproducible example [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

Feel free to reach out if you have any more questions. Happy modeling! 🚀

**Comment by pornpra** [comment]
@glenn-jocher 

Currently, I am using Ultralytics version 8.2.18 and have tried to convert a custom YOLOv8 classification model from FP32 to INT8 using this command

`yolo export model=yolov8s_car_brand_fp32_25_07_2024.pt format=engine device=0 int8=True workspace=4 data=car_brand.yaml
`

, but I got this error:

![yolov8_error](https://github.com/user-attachments/assets/f517521c-21c5-40d6-ad61-70bc7be7946c)

Inside car_brand.yaml, I have already defined the path and list of image files in .txt format for both the training set and test set as follows:

path: /home/nvidia/ORDS/int8/car_brand/yolov8 
train: train_car_brand.txt  
val: test_car_brand.txt  

Can you help me solve this export issues ?
